app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt

```diff
@@ -110,12 +107,20 @@ class UserTokensPolling(
     private suspend fun fetchSolTokens(): List<Token.Active> =
         userInteractor.loadUserTokensAndUpdateLocal(tokenKeyProvider.publicKey.toPublicKey())
 
-    private suspend fun fetchEthereumTokens(): List<Token.Eth> {
-        if (!ethAddressEnabledFeatureToggle.isFeatureEnabled) {
-            return emptyList()
+    private suspend fun fetchEthereumTokens() {
+        if (ethAddressEnabledFeatureToggle.isFeatureEnabled) {
+
+            val ethBundles = ethereumInteractor.getListOfEthereumBundleStatuses()
+            ethereumInteractor.loadWalletTokens(ethBundles)
+        }
+    }
+
+    private fun getEthereumTokensFlow(): Flow<List<Token.Eth>> {
+        return if (ethAddressEnabledFeatureToggle.isFeatureEnabled) {
+            return ethereumInteractor.getTokensFlow()
```

Comment:
no need for return

Reply:
maybe you can extract using ethAddressEnabledFeatureToggle inside interactor to remove duplicates?